examples/foundational/10-wake-word.py

```diff
@@ -78,6 +85,33 @@
 thinking_frame = SpriteFrame(thinking_list)
 
 
+class NameCheckFilter(FrameProcessor):
+    def __init__(self, names: list[str]):
+        super().__init__()
+        self._names = names
+        self._sentence = ""
+
+    async def process_frame(self, frame: Frame, direction: FrameDirection):
+        if isinstance(frame, SystemFrame):
+            await self.push_frame(frame, direction)
+            return
+
+        content: str = ""
+
+        # TODO: split up transcription by participant
+        if isinstance(frame, TranscriptionFrame):
+            content = frame.text
+            self._sentence += content
+            if self._sentence.endswith((".", "?", "!")):
+                if any(name in self._sentence for name in self._names):
+                    await self.push_frame(TextFrame(self._sentence))
+                    self._sentence = ""
+                else:
+                    self._sentence = ""
+        else:
+            await self.push_frame(frame, direction)
+
+
```

Comment:
Watch out with re-adding this. We now have `WakeCheckFilter` which supersedes `NameCheckFilter`. I think you just need the 2 VAD lines.

Reply:
And the 2 `LLMUserResponseAggregator` and assistant lines.